dataeng/resources/tableau-config-backup.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -ex
+
+ssh -tt $USER_NAME@$TABLEAU_SERVER_IP -o StrictHostKeyChecking=no "S3_BUCKET=$S3_BUCKET" '
+
+# It gets the current date from system in 2022.04.05-10:35 format
+DATE="$(date '+%Y.%m.%d')"
+
+# It adds the date as prefix on backup name. Backup runs at 09:00 UTC so it also prefixed in the name.
+BACKUPNAME="${DATE}-tableau-config-backup.json"
+
+sudo runuser -l tsm_admin -c "tsm settings export -f /home/tableau_backups/tableau-config-backup.json"
+
+# Check if tableau backup file exist, exit if the file does not exist
+if [[ ! -f /home/tableau_backups/tableau-config-backup.json ]]
+then
+    echo "$BACKUPNAME Backup file does not exist. It seems to be some problem generating tableau backup. Refer to Administration of Tableau wiki docs to debug Tableau Regular Backup Automation"
+    exit 1
+else
+    echo "Copying backup file into s3"
+    aws s3 cp /home/tableau_backups/tableau-config-backup.json s3://$S3_BUCKET/$BACKUPNAME
+
+    echo "Removing backup file from local disk"
+    sudo rm -f /home/tableau_backups/$BACKUPNAME
+
+fi
+
+'
+
+# Creating python 3.8 virtual environment to run python script.
+PYTHON38_VENV="py38_venv"
+virtualenv --python=python3.8 --clear "${PYTHON38_VENV}"
+source "${PYTHON38_VENV}/bin/activate"
+
+DATE="$(date '+%Y.%m.%d')"
+TABLEAU_CONFIG=${DATE}-tableau-config-backup.json
+
+# Copying tableau-config-backup file from s3.
+aws s3 cp s3://$S3_BUCKET/$TABLEAU_CONFIG tableau-config.json
+
+# Copying tableau-config-json-serializer from analytics-tolls repository to local repo.
+cp $WORKSPACE/analytics-tools/tableau/tableau-config-json-serializer.py tableau-config-json-serializer.py
+
+# This script deletes the json object which shows non-relevant change.
+python tableau-config-json-serializer.py
+
+# Comparing both base line config file with the latest backup obtained from tableau.
+if cmp -s "$WORKSPACE/analytics-secure/tableau/tableau-config-backup-base.json" "serialized-tableau-config.json"; then
+    echo "Tableau Config Backups are same"
+else
+    # Changes are detected in Tableau config so it will create a pull request to update baseline backup file.
+    echo "Tableau Config Backups are modified"
+
+    cp serialized-tableau-config.json $WORKSPACE/analytics-secure/tableau/tableau-config-backup-base.json
+
+    cd $WORKSPACE/analytics-secure/
+
+    # Create a new git branch
+    now=$(date +%Y_%m_%d_%H_%M)
+    branchname="tableau_config_$now"
```

Comment:
nit: can you standardize on CAPs for bash variables?

Reply:
Thanks, fixed it.